src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -390,7 +384,7 @@ internal static class DependentTypeFinder
                                                  .ToList();
         }
 
-        private static async Task<IEnumerable<INamedTypeSymbol>> FindMetadataTypesInProjectAsync(
+        private static async Task<IEnumerable<INamedTypeSymbol>> FindAllMatchingMetadataTypesInProjectAsync(
             HashSet<INamedTypeSymbol> metadataTypes,
```

Comment:
Yes, it's the input.  This is part of a larger component/module that is responsible for starting with a type and cascading (immediately or transitively) to 'dependent' types that have some sort of relationship with it (generally 'derived in the case of classes, and 'derived or implementing classes' in the case of interfaces).

Because we're cascading, we might start with a single type, then branch and find many types that have the right relationship.   The transitive algorithms work until a fixed point is reached.

For example, start with IDisposable (a metadata type).  In metadata we'll find derived interfaces of that type, and classes/structs that implement that type.  We'll then search for type that have the requested relationship with that set, and so on and so on.